Update dev guide (#96)
AY2122S1-CS2103T-W16-2/tp: docs/DeveloperGuide.md

### Use cases

(For all use cases below, the **System** is the `BobTheBistroBoss (BTBB)` and the **Actor** is the `Home Chef`, unless specified otherwise)

**Use case: UC01 Add a client**

**MSS**

1.  Home Chef chooses to add a client.
2.  Home Chef enters details in required format.
3.  BTBB displays new client.

    Use case ends.

**Extensions**

* 2a. Phone number entered already exists in the system.
    * 2a1. BTBB informs Home Chef that client already exist.

      Use case ends.

* 2b. BTBB detects an error in the entered data.

>     * 2b1. BTBB informs Home Chef why adding did not happen.

Reviewer: ```suggestion
    * 2b1. BTBB informs Home Chef why addition of client failed.
```

      Use case ends.

**Use case: UC02 Find a client**

**MSS**

1.  Home Chef wants to find a client by a field.
2.  Home Chef enters search terms in required format.

> 3.  BTBB displays all clients that match the format.

Reviewer: ```suggestion
3.  BTBB displays all clients that match the terms.
```

    Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs Home Chef why find did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs Home Chef why finding a client failed.
```

      Use case ends.

**Use case: UC03 Delete a client**

**MSS**

1.  Home Chef chooses to delete a client.
2.  Home Chef enter details to delete a client.

> 3.  BTBB deletes client.

Reviewer: Remove this line

4.  BTBB informs Home Chef that client has been deleted.

    Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs Home Chef why deletion did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs Home Chef why deletion of client failed.
```

      Use case ends.

**Use case: UC04 View all clients**

**MSS**

1.  Home Chef chooses to view all clients.
2.  Home Chef enters command to see all clients.
3.  BTBB displays all clients.

    Use case ends.

**Use case: UC05 Edit clients**

**MSS**

1.  Home chef chooses to edit a client’s details.
2.  Home chef enters details in required format.
3.  BTBB displays the edited client.

    Use case ends.

**Extensions**

* 2a. Phone number entered already exists in the system.
    * 2a1. BTBB informs user that client already exist.

      Use case ends.

* 2b. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why editing of client failed.
```

      Use case ends.

**Use case: UC06 Add ingredient**

**MSS**

1.  Home chef chooses to add ingredient.
2.  Home chef enters details in required format.
3.  BTBB displays new ingredient.

    Use case ends.

**Extensions**

* 2a. Ingredient entered already exists in the system.
    * 2a1. BTBB informs user that ingredient already exist.

      Use case ends.

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why addition of ingredient failed.
```

      Use case ends.

**Use case: UC07 Find Ingredient**

**MSS**

1.  Home chef wants to find an ingredient by a field.
2.  Home chef enters search terms in required format.
3.  BTBB displays all ingredient that match the format.

    Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why find did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why finding an ingredient failed.
```

    Use case ends.

**Use case: UC08 Delete an Ingredient**

**MSS**

1.  Home chef chooses to delete a ingredient.
2.  Home chef enters details in required format.
3.  BTBB displays deleted ingredient.

    Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why deleting did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why deletion of ingredient failed.
```

      Use case ends.

**Use case: UC09 View all ingredients**

**MSS**

1.  Home chef chooses to view all ingredients.
2.  Home chef enters command to see all ingredients.
3.  BTBB displays all ingredients.

    Use case ends.

**Use case: UC10 Edit Ingredient**

**MSS**

1. Home chef chooses to edit an ingredient.
2. Home chef enters details in required format.
3. BTBB displays the edited ingredient.

    Use case ends.

**Extensions**

* 2a. Ingredient entered already exists in the system.
    * 2a1. BTBB informs user that ingredient already exist.

      Use case ends.

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why editing of ingredient failed.
```

      Use case ends.

**Use case: UC11 Add Order**

**MSS**

1. Home chef chooses to add order.
2. Home chef enters details in required format.
3. BTBB displays new order.

    Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why addition of order failed.
```

    Use case ends.

**Use case: UC12 Delete Order**

**MSS**

1. Home chef chooses to delete an order.
2. Home chef enters details in required format.
3. BTBB displays deleted order.

   Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why deleting did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why deletion of order failed.
```

      Use case ends.

**Use case: UC13 Find order**

**MSS**

1. Home chef chooses to find an order.
2. Home chef enters details in required format.
3. BTBB displays all orders that match the format.

   Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why find did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why finding an order failed.
```

      Use case ends.

**Use case: UC14 Mark order as done**

**MSS**

1. Home chef chooses to mark an order as done.
2. BTBB displays order as done.

   Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why marking an order as done failed.
```

      Use case ends.

**Use case: UC15 Mark order as undone**

**MSS**

1. Home chef chooses to mark an order as undone.
2. BTBB displays order as undone.

   Use case ends.

**Extensions**

* 2a. BTBB detects an error in the entered data.

>     * 2a1. BTBB informs user why adding did not happen.

Reviewer: ```suggestion
    * 2a1. BTBB informs user why marking an order as undone failed.
```
